projects/cdk/utils/miscellaneous/is-valid-url.ts

```diff
@@ -0,0 +1,13 @@
+export function tuiIsValidUrl(url: string): boolean {
+    const pattern = new RegExp(
+        `^([a-zA-Z]+:\\/\\/)?` + // protocol
+            `((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|localhost|` + // domain name
+            `((\\d{1,3}\\.){3}\\d{1,3}))` + // OR IP (v4) address
+            `(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*` + // port and path
+            `(\\?[;&a-z\\d%_.~+=-]*)?` + // query string
+            `(\\#[-a-z\\d_]*)?$`, // fragment locator
```

Comment:
I can't validate it, I just hope it works :D

Reply:
unit tests for help us)

Reply:
Chat GPT to the rescue:
![image](https://user-images.githubusercontent.com/11832552/235856577-aa631101-4a9e-4fb1-9cf4-16e649afd9c5.png)